extensions/livedata/src/main/java/com/twocoders/extensions/livedata/LiveDataExtensions.kt

```diff
@@ -0,0 +1,382 @@
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+
+fun <T : Any?> LiveData<T>.asLiveEvent(): LiveEvent<T> =
+    if (this is LiveEvent<T>) this else throw IllegalArgumentException("$this is not an instance of LiveEvent!")
+
+fun <T : Any?> LiveData<T>.asMutable(): MutableLiveData<T> =
+    if (this is MutableLiveData<T>) this else throw IllegalArgumentException("$this is not an instance of MutableLiveData!")
+
+fun LiveData<String>.isEmpty(): Boolean = value?.isEmpty() ?: true
+
+fun LiveData<String>.isNotEmpty() = !isEmpty()
+
+fun LiveData<Boolean>.isTrue() = value == true
+
+fun LiveData<Boolean>.isNotTrue() = !isTrue()
+
+fun MutableLiveData<Boolean>.toggle() {
+    value = !value!!
+}
+
+fun <T : Any?> MutableLiveData<T>.notifyObservers() {
+    this.value = this.value
+}
+
+fun <K, V> MutableLiveData<Map<K, V>>.put(key: K, newValue: V) {
+    (this.value as MutableMap)[key] = newValue
+    notifyObservers()
+}
+
+@MainThread
+fun <T> LiveData<T>.observeOnce(observer: (T) -> Unit) {
+    observeForever(object : Observer<T> {
+        override fun onChanged(value: T) {
+            removeObserver(this)
+            observer(value)
+        }
+    })
+}
+
+@MainThread
+fun <T> LiveData<T>.observeOnce(owner: LifecycleOwner, observer: (T) -> Unit) {
+    observe(owner, object : Observer<T> {
+        override fun onChanged(value: T) {
+            removeObserver(this)
+            observer(value)
+        }
+    })
+}
+
+@MainThread
+fun <X> LiveData<X>.filter(test: (X) -> Boolean): LiveData<X> {
+    val result = MediatorLiveData<X>()
+
+    result.addSource(this) {
+        if (it != null && test(it)) {
+            result.value = it
+        }
+    }
+
+    return result
+}
+
+@MainThread
+fun <X> LiveData<X>.filterNullables(test: (X?) -> Boolean): LiveData<X> {
+    val result = MediatorLiveData<X>()
+
+    result.addSource(this) {
+        if (test(it)) {
+            result.value = it
+        }
+    }
+
+    return result
+}
+
+@MainThread
+fun <X> LiveData<X?>.filterNull(): LiveData<X> {
+    val result = MediatorLiveData<X>()
+
+    result.addSource(this) {
+        it?.let {
+            result.value = it
+        }
+    }
+
+    return result
+}
+
+@MainThread
+fun <X> LiveData<X>.take(count: Int): LiveData<X> {
+    var taken = 0
+    val result = MediatorLiveData<X>()
+
+    result.addSource(this) {
+        it?.let {
+            if (taken < count) {
+                result.value = it
+                taken++
+            }
+
+            if (taken >= count) {
+                result.removeSource(this)
+            }
+        }
+    }
+
+    return result
+}
+
+@MainThread
+fun <X> LiveData<X>.skip(count: Int): LiveData<X> {
+    var skipped = 0
+    val result = MediatorLiveData<X>()
+
+    result.addSource(this) {
+        it?.let {
+            if (skipped >= count) {
+                result.value = it
+            }
+
+            if (skipped < count) {
```

Comment:
`else` will do the same as the separate new `if`
```suggestion
            } else {
```